odoo/addons/base/ir/ir_autovacuum.py

```diff
@@ -20,7 +20,7 @@ def _gc_transient_models(self):
             if model.is_transient():
                 try:
                     with self._cr.savepoint():
-                        model._transient_vacuum(force=True)
+                        model._transient_vacuum(force=True, autocommit=True)
```

Comment:
This will not work: as soon as you commit the cursor inside `model._transient_vacuum`, the savepoint set by `self._cr.savepoint()` is gone.

Reply:
Indeed, I didn't consider this savepoint.

So I see 2 workarounds:
- only commit between models (crappy)
- commit at the start of _gc_transient_models and replace the savepoint by a try-except with a sefl._cr.rollback().

Ideally _gc_transient_models should have an autocommit parameter itself and do the old way & new way depending on the parameter. This is likely going to be very ugly.

Do you have a better idea?

I'm considering to drop the autocommit fix due to this and only keep the memory error fix.